pep-NNN.rst

```diff
@@ -84,33 +162,54 @@ Specification
    other semantically different version numbers (e.g. underlying library
    version number) consistent with PEP 440.
 5. The ``__version__`` attribute SHOULD be consistent with the ``importlib.metadata``
-   information.
+   information, where applicable.
 6. Packages MAY dynamically compute the ``__version__`` string at runtime.
 
+NOTE: this PEP does not propose the use of ``__version__`` in any standard library modules.
 
 Backwards Compatibility
 =======================
 
 [Describe potential impact and severity on pre-existing code.]
 
-None.  This is already a well established convention
+This is already a well established convention -- there are three possible conditions of currently available packages:
+
+1. It already has a ``__version__`` attribute: no impact.
+
+2. It doesn't have a version attribute at all: these packages would need to add the attribute to be compliant, which would be fully backward compatible.
+
+3. It has an attribute with a different name (e.g. ``VERSION``), or in a different location, such as a ``version.py`` file. In this case, the package would need to add a ``__version__`` attribute to be compliant, but could maintain an alias in the old name to ease the transition to the new format (and keep that alias indefinitely, if desired).
+
```

Comment:
Hmm -- it a way, the PEP 440 issue is a bit orthogonal to this proposal -- I do think that it's a good idea to include it -- why standardize only part of the process?

But I don't think the backward compatibility is too bad -- if a project uses non-standard versioning scheme, they can continue to do so, and won't quite be fully in compliance with this PEP. But they are already not compliant with PEP 440, so is it a big deal?

I suppose they could be using a PEP 440  compliant version string for the distribution package, but not the __version__ string?

Should we add that to the list of potential backward compatibility issues?

Reply:
I think it should be added to the list of potential backward compatibility issues, just because it seems like an omitted corner case otherwise. And I likewise don't want to drop the core recommendation at the top that `__version__` should be a PEP 440 string.

But I like that you suggest that "doing nothing wouldn't make things any worse" here.

Maybe something like:

```
It has an existing `__version__` attribute, but its value is not a PEP 440-conforming string. We expect this case to be exceedingly rare, and already puts the package in a special case where PEP 440 non-compliance is already being worked around.
```

Maybe this is pushing us to an overall spec of "All packages *MUST* have a __version__, and it *SHOULD* be PEP-440 compliant" -- or maybe this is too much of a special case to turn that second *MUST* into a *SHOULD*.  (And I know overall we are withholding finalizing that until we get more feedback anyway...)

Reply:
Thanks -- I've added that text, and changed one SHOULD to a MUST, but not the other :-)